caddyhttp/basicauth/basicauth.go

```diff
@@ -63,7 +63,7 @@ func (a BasicAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 
 			// let upstream middleware (e.g. fastcgi and cgi) know about authenticated
 			// user; this replaces the request with a wrapped instance
-			r = r.WithContext(context.WithValue(r.Context(),
+			*r = *r.WithContext(context.WithValue(r.Context(),
```

Comment:
@mholt @tobya just an FYI,  For the past 9 days, I've been running in production a build of Caddy with this PR cherry-picked and my own auth module that [closely copies the `basicauth` ](https://github.com/jamesboswell/caddy-radius/compare/caddyPR1542) changes in this PR.

Caddyfile is now using the updated `{combined}` log format instead of my previous custom format.

So far everything looks great in regards to logging `{user}` in access logs.